CodeEdit/Features/Editor/PathBar/Views/EditorPathBarComponent.swift

```diff
@@ -58,11 +62,52 @@ struct EditorPathBarComponent: View {
             button.font = .systemFont(ofSize: NSFont.systemFontSize(for: .small))
             button.isBordered = false
             (button.cell as? NSPopUpButtonCell)?.arrowPosition = .noArrow
+            DispatchQueue.main.async {
```

Comment:
This feels a bit weird. Can't this be done some other way instead of storing a reference to the NSPopupButton?

Reply:
I do agree, this feels odd.

Also does something like this not work (if you put it in the `init()` function):
```swift
self.button = {
    let button = NSPopUpButton()
    button.menu = EditorPathBarMenu(
        fileItems: siblings,
        fileManager: fileManager,
        tappedOpenFile: tappedOpenFile
    )
    button.font = .systemFont(ofSize: NSFont.systemFontSize(for: .small))
    button.isBordered = false
    (button.cell as? NSPopUpButtonCell)?.arrowPosition = .noArrow

    return button
}()
```

Reply:
This was @thecoolwinter's recommended approach. I don't think I have access to `fileManager` in the `init` function.

Reply:
How should we proceed here?

Reply:
@Wouter01 Okay, your implementation you shared on discord worked. Give that a look and approve if that is good.